### ISSUE TITLE
Monitoring pod change name pattern after restart

### ISSUE THREAD

**Issue by alirezaDavid** [rook/rook#850]
I deployed rook cluster for a month but after a restart, I think monitoring pod changes their name pattern.
```
mon6-rtnz2                        1/1       Running   0          5d
mon8-6v1tl                        1/1       Running   0          5d
rook-api-2485995279-4c6tv         1/1       Running   0          9d
rook-ceph-mds-3563284940-057nb    1/1       Running   0          9d
rook-ceph-mgr0-2234418690-nvqgr   1/1       Running   0          9d
rook-ceph-mon1-w44zc              1/1       Running   0          9d
rook-ceph-osd-worker-03-6msf4     1/1       Running   0          9d
rook-ceph-osd-worker-04-26577     1/1       Running   0          9d
rook-tools                        1/1       Running   0          9d
```
For a first time deploy all monitoring pod names pattern was like this `rook-ceph-mon1-*` but I think after restart their name changed.

**Comment by travisn** [comment]
Hi @alirezaDavid, thanks for reporting, there is a fix for this issue with #806.  There shouldn't be any side effects other than the naming consistency issue.

**Comment by alirezaDavid** [comment]
@travisn Thanks 👍
